### DISCUSSION TITLE
Model with script tag does not work

### DISCUSSION THREAD

**Discussion by jeroenvanrensen** [alpinejs/alpine#1343] [1. Help] [answered]
Hi everyone,

I created this Alpine.js component:

```
<div>
    <div x-data="tagInput()">
        <input type="text" x-model="tagName" @keydown.space="addTag" />
    </div>
    
    <script>
        function tagInput() {
            return {
                tagName: '',

                addTag() {
                    console.log(this.tagName);
                    this.tagName = '';
                }
            }
        }
    </script>
</div>
```

This is what it should do:

1. A user types in a tag
2. The user hits space
3. The tag name is in the console
4. The input gets cleared out

But step 4 does not work. It does not clear. I also don't get an error.

Here's a demo: https://www.w3schools.com/code/tryit.asp?filename=GQW162CRP251

Thank you!

**Comment by jeroenvanrensen** [comment, marked as answer]
It already got answered here: https://laracasts.com/discuss/channels/javascript/alpine-js-model-with-script-tag-does-not-work

**Reply by SimoTod** [reply]
the technical reason is that you are setting the variable to empty on keydown but the standard x-model event in Alpine triggers on input and the input event fires after the keydown one so the model event would set again the previous value on the tagName variable.

**Reply by jeroenvanrensen** [reply]
@SimoTod thanks for your detailed explanation, now I understand!
